### PR TITLE
Fix `test_gather`

### PR DESCRIPTION
**Pull request by ESI-SYD**
Make `getStackPointer` as interface of the `TargetInfo` to generalize `getSharedMemoryBase` in gather op.

### CODE REVIEW

**Comment by chengjunlu** [comment]
The major difference of the lowering gather operation is `intel::LLVM::getSharedMemoryBase`. 
We can make the function `LLVM::intel::getStackPointer` to be a interface of the `TargetInfo`.

Can we unify the lowering code to the upstream Triton?

**Comment by ESI-SYD** [comment]
> The major difference of the lowering gather operation is `intel::LLVM::getSharedMemoryBase`. We can make the function `LLVM::intel::getStackPointer` to be a interface of the `TargetInfo`.
> 
> Can we unify the lowering code to the upstream Triton?

Yes, we have different `getStackPointer` from public, for this op, this PR inherited and implemented, followed similar logic of `ControlFlow` op. [code](https://github.com/intel/intel-xpu-backend-for-triton/blob/main/third_party/intel/lib/TritonIntelGPUToLLVM/ControlFlowOpToLLVM.cpp#L89)

So if we want to add `getStackPointer` as interface of the `TargetInfo`, and we should also consider `ControlFlow` op lowering.

Because NV and AMD shared same `getStackPointer`, I'm not sure whether it's acceptable if we introduce such changes for upstream.

**Comment by whitneywhtsang** [comment]
Please try to upstream these changes.

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h [hunk 1]
```diff
@@ -83,35 +83,21 @@ Value addStringToModule(Location loc, RewriterBase &rewriter, StringRef key,
...
-                             FunctionOpInterface funcOp) {
-  auto mod = funcOp->getParentOfType<ModuleOp>();
-  LLVM::LLVMPointerType ptrTy = ptr_ty(
-      rewriter.getContext(), TritonGEN::TritonGENMemorySpace::kWorkgroup);
-  if (mod->getAttrOfType<IntegerAttr>("ttg.shared").getInt() == 0)
-    return rewriter.create<LLVM::PoisonOp>(funcOp.getLoc(), ptrTy);
-  return funcOp.getArgument(funcOp.getNumArguments() - 1);
-}
-
 static Value getSharedMemoryBase(Location loc,
```
Comment (chengjunlu) [inline third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h:86]: Can we remove this since it is same to the one in upstream Utility.h

Comment (ESI-SYD) [reply third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h]: yes

**Review by chengjunlu — APPROVED** [review]

**Comment by ESI-SYD** [comment]
> Please try to upstream these changes.

https://github.com/triton-lang/triton/pull/5451

**Pull request merged**